composer/models/ssd/ssd.py

```diff
@@ -0,0 +1,173 @@
+        input_size = 300
+        val_trans = SSDTransformer(dboxes, (input_size, input_size), val=True)
+        data = "/localdisk/coco"
+
+        val_annotate = os.path.join(data, "annotations/instances_val2017.json")
+        val_coco_root = os.path.join(data, "val2017")
+        train_annotate = os.path.join(data, "annotations/instances_train2017.json")
+        train_coco_root = os.path.join(data, "train2017")
+        val_coco = COCODetection(val_coco_root, val_annotate, val_trans)
+
+        inv_map = {v: k for k, v in val_coco.label_map.items()}
+
+        ret = []
+
+        overlap_threshold = 0.50
+        nms_max_detections = 200
+
+        dboxes = dboxes300_coco()
+        encoder = Encoder(dboxes)
+
+        (img, img_id, img_size, _, _) = batch
+        targets = get_boxes(val_annotate, img_id)
+
+        ploc, plabel = self.module(img)
+        ploc, plabel = ploc.float(), plabel.float()
+
+        for idx in range(ploc.shape[0]):
+            ploc_i = ploc[idx, :, :].unsqueeze(0)
+            plabel_i = plabel[idx, :, :].unsqueeze(0)
+
+            try:
+                result = encoder.decode_batch(ploc_i, plabel_i, overlap_threshold, nmx_max_detections)[0]
+            except:
+                # raise
+                print("")
+                print("No object detected in idx: {}".format(idx))
+                continue
+
+            htot, wtot = img_size[0][idx].item(), img_size[1][idx].item()
+            loc, label, prob = [r.cpu().numpy() for r in result]
+
+            for loc_, label_, prob_ in zip(loc, label, prob):
+
+                ret.append([
+                dict(
+                        boxes=torch.Tensor([[loc_[0] * wtot, \
+                                             loc_[1] * htot,
+                                             (loc_[2] - loc_[0]) * wtot,
+                                             (loc_[3] - loc_[1]) * htot]]),
+                        scores=torch.Tensor([prob_]),
+                        labels=torch.IntTensor([inv_map[label_]]),
+                    )
+                ])
+
+        return ret, targets
+
+
+def get_boxes(val_annotate, idss):
+    ids = idss.tolist()
+    from pycocotools.coco import COCO
+    our_coco = COCO(annotation_file=val_annotate)
+    import json
+    json_file = "/localdisk//coco/annotations/instances_val2017.json"
+
+    with open(json_file, 'r') as COCO:
+        js = json.loads(COCO.read())
+        anns_all = json.dumps(js['annotations'])
+
+    annids = our_coco.getAnnIds(imgIds=ids)
+    anns = our_coco.loadAnns(annids)
+
+    t = []
+    for ann in anns:
+        x_topleft = ann['bbox'][0]
+        y_topleft = ann['bbox'][1]
+        bbox_width = ann['bbox'][2]
+        bbox_height = ann['bbox'][3]
+
+        cat = ann['category_id']
+
+        t.append(dict(boxes=torch.Tensor([[x_topleft, y_topleft, bbox_width, bbox_height]]),
+                      labels=torch.Tensor([cat])))
+
+    #
+    return t
+
+
+def dboxes300_coco():
```

Comment:
In addition to Austin's comment, is this the same as the dboxes300_coco in the dataset code? If so, delete this duplicate, but apply Austin's comment to the other code.